loaders/src/glTF/2.0/babylon.glTFLoader.ts

```diff
@@ -80,20 +80,38 @@ module BABYLON.GLTF2 {
         }
 
         // IE 11 Compatibility.
-        private static _createProgressEvent: (lengthComputable: boolean, loaded: number, total: number) => ProgressEvent =
-            (typeof (<any>window)["ProgressEvent"] === "function")
-                ? (lengthComputable, loaded, total) => {
+        private static _createProgressEvent = GLTFLoader._getCreateProgressEventFunction();
+        private static _getCreateProgressEventFunction(): (lengthComputable: boolean, loaded: number, total: number) => ProgressEvent {
+            // Use ProgressEvent if available.
+            if (typeof (<any>window)["ProgressEvent"] === "function") {

```

Comment:
beware as this won't work in nullengine mode as window is not defined

Reply:
is there a better way to check the existence of ProgressEvent class?

Reply:
you can first call Tools.IsWindowObjectExist()
as a side node: why do we need to rely on a DOM element and not on our own object?

Reply:
this is supposed to match what XHR does, but maybe it's okay to change to our own object? the ISceneLoaderPluginAsync interface is currently defined to return a ProgressEvent for onProgress.

Reply:
We can change that easily. My concern is that we now have NulLEngine in placed and used by several customers. This will break the nullengine scenario

Reply:
Okay, I'll make this change and update.

Reply:
much appreciated:)